daemon/algod/api/server/v1/handlers/handlers.go

```diff
@@ -207,6 +257,40 @@ func assetConfigTxEncode(tx transactions.Transaction, ad transactions.ApplyData)
 	}
 }
 
+func applicationCallTxEncode(tx transactions.Transaction, ad transactions.ApplyData) v1.Transaction {
+	b64 := base64.StdEncoding
+	app := v1.ApplicationCallTransactionType{
+		ApplicationID:     uint64(tx.ApplicationID),
+		ApprovalProgram:   b64.EncodeToString([]byte(tx.ApprovalProgram)),
+		ClearStateProgram: b64.EncodeToString([]byte(tx.ClearStateProgram)),
+		LocalStateSchema:  modelSchema(tx.LocalStateSchema),
+		GlobalStateSchema: modelSchema(tx.GlobalStateSchema),
+		OnCompletion:      tx.OnCompletion.String(),
+	}
+
+	var encodedAccounts []string
```

Comment:
nit: preallocate `encodedAccounts`, `encodedForeignApps` and `encodedForeignApps`

Reply:
done, thanks!